### PR TITLE
Inception Time model for time series classification

### PR DESCRIPTION
**Pull request by Christosc96**

### CODE REVIEW

**Review by dhavalsalwala — CHANGES_REQUESTED** [review]
Hi @Christosc96 ,
Thanks for submitting the PR.

Can you please rename the test file to test_inceptiontime.py? Also, if you have a demo notebook, could you upload it to sail/notebooks?

**Review by dhavalsalwala — CHANGES_REQUESTED** [review]
Hi @Christosc96 ,

Your unit test is failing. Could you please check?
tests/models/torch/test_inceptiontime.py

File: sail/models/torch/inceptiontime.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    
+    super(InceptionTimeClassifier, self).__init__(
+            module=self.inception_model,
+            max_epochs=max_epochs,
+            lr=learning_rate,
+            batch_size=batch_size,
+            criterion=criterion ,
+            # Shuffle training data on each epoch
+            iterator_train__shuffle=False,
+            device='cuda',
```
Comment (dhavalsalwala) [inline sail/models/torch/inceptiontime.py]: Make default value to cpu.

**Comment by Christosc96** [comment]
Hi @dhavalsalwala, I noticed I the classifier import statement was incorrect and fixed it. The tests do not fail when I try to run them locally (I provide the data in the test file though through a method which I do not include here to be consistent with the reference test file for the rnn classifier).

File: tests/models/torch/test_inceptiontime.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            net.partial_fit(X, y)
+        train_losses = net.history[:, 'train_loss']
+        assert train_losses[0] > train_losses[-1]
+
+    def test_predict(self, net_partial_fit, classification_x, classification_y):
+        X = classification_x
+        y_pred = net_partial_fit.predict(X)
+
+        # predictions should not be all zeros
+        assert not np.allclose(y_pred, 0)
```
Comment (dhavalsalwala) [inline tests/models/torch/test_inceptiontime.py:57]: np is not defined.

File: tests/models/torch/test_inceptiontime.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+'''
+
+class TestInceptionTime:
+    @pytest.fixture
+    def net(self):
+        from  sail.models.torch.inceptiontime import InceptionTimeClassifier
+        return InceptionTimeClassifier(lr=0.07)
+
+    @pytest.fixture
+    def net_partial_fit(self, net, classification_x, classification_y):
```
Comment (dhavalsalwala) [inline tests/models/torch/test_inceptiontime.py]: classification_x is not defined.

File: tests/models/torch/test_inceptiontime.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+'''
+UNIT TESTING
+'''
+
+class TestInceptionTime:
+    @pytest.fixture
+    def net(self):
+        from  sail.models.torch.inceptiontime import InceptionTimeClassifier
+        return InceptionTimeClassifier(lr=0.07)
```
Comment (dhavalsalwala) [inline tests/models/torch/test_inceptiontime.py]: HI @Christosc96 ,
As per your code, it should be learning_rate instead of lr.

**Comment by dhavalsalwala** [comment]
> Hi @dhavalsalwala, I noticed I the classifier import statement was incorrect and fixed it. The tests do not fail when I try to run them locally (I provide the data in the test file though through a method which I do not include here to be consistent with the reference test file for the rnn classifier).

You need to define your dataset(classification_x) in tests/models/torch/conftest.py. If possible, use synthetic dataset for unit tests or use pre-availables from scikit-learn.

Examples:
Synthetic dataset: tests/models/torch/conftest.py:regression_data()
scikit-learn dataset: tests/models/river/test_linear_model.py classification_dataset(self)

**Comment by Christosc96** [comment]
I used the same dataset i used in the notebook, InceptionTime requires a bit of editing to data in order to be compatible with expected model dimensions so I put the dataset as a fixture in inception-time's own test file

**Review by dhavalsalwala — APPROVED** [review]
Looks Good to me.

**Pull request merged**